### ISSUE TITLE
XML encoding property

### ISSUE THREAD

**Issue by IhorKirei** [jhuckaby/pixl-xml#8]
Hello there!

Is it possible to setup encoding property for XML document?

Like that:
`<?xml version="1.0" encoding="UTF-8"?>`

**Comment by jhuckaby** [comment]
I'm sorry to say, this currently isn't possible using the documented API.  There technically IS a way, but it uses a hidden object-oriented API that is totally undocumented.  I'll briefly explain it here, but it's unsupported.

If you parse an XML document using this method, it will preserve the PI (Processor Instruction) node at the top of the file:

```js
var PixlXML = require('pixl-xml');

var xml_string = '<?xml version="1.0" encoding="UTF-8"?><Document>' + 
	'<Simple>Hello</Simple>' + 
	'<Node Key="Value">Complex</Node>' + 
	'</Document>';

var parser = new PixlXML.XML( xml_string );
var doc = parser.getTree();

doc.Simple = "Hello, I changed this.";

console.log( parser.compose() );
```

This will output the following:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<Document>
	<Node>Complex</Node>
	<Simple>Hello, I changed this.</Simple>
</Document>
```

You can also change the PI node by setting it manually on the parser object before you call `compose()`:

```js
parser.piNodeList = [ '?xml version="1.0" encoding="UTF-8"?' ];
```

Make sure to include everything inside the angle brackets, but not the brackets themselves.

I'm very sorry this isn't documented, but you are free to use this if you want.  Just please note, this is at your own risk.

**Comment by jhuckaby** [comment]
I just released [Version 1.0.8](https://github.com/jhuckaby/pixl-xml/commit/42ec1db0260ba65c20de1ce4f82b213fa37da33a) which has an officially documented object-oriented API, that allows you to have more control over the serialization (specifically, you can control the PI and DTD nodes).  Hope this helps!

Docs copypasta:

# Object-Oriented API

In addition to the [Simplified API](https://github.com/jhuckaby/pixl-xml#simplified-api), an object-oriented API is also available.  Using this, you instantiate an `XML.Parser` class instance, and use that to parse, manipulate and serialize XML.  The constructor accepts up to two arguments, the raw XML string, and an optional object with configuration options.

After constructing the `XML.Parser` object, and no error was thrown, call `getTree()` to get a reference to the simplified XML structure in memory, manipulate it if you want, then call `compose()` to serialize the object tree back into XML.

The main reason for using this API is that it preserves any PI (Processing Instructions) and DTD (Document Type Definition) elements in the source XML file, and they will be serialized into the output.  Example:

``` js
var xml_string = '<?xml version="1.0" encoding="UTF-8"?><Document>' + 
    '<Simple>Hello</Simple>' + 
    '<Node Key="Value">Complex</Node>' + 
    '</Document>';

var parser = null;
try {
    parser = new XML.Parser( xml_string, { preserveAttributes: true } );
}
catch (err) {
    throw err;
}

var doc = parser.getTree();
doc.Simple = "Hello, I changed this.";

console.log( parser.compose() );
```

This would produce the following output:

``` xml
<?xml version="1.0" encoding="UTF-8"?>
<Document>
    <Node Key="Value">Complex</Node>
    <Simple>Hello, I changed this.</Simple>
</Document>
```

Notice that the PI element with its `encoding` attribute was preserved and serialized.

To manipulate the PI nodes, access the `piNodeList` property.  It is an array of strings, each one representing one raw PI node sans the surrounding angle brackets, e.g. `?xml version="1.0"?`.  Similarly, any DTD nodes are stored in the `dtdNodeList` property (also an array).  Feel free to change these to customize your serialized XML document.

``` js
parser.piNodeList = [ '?xml version="1.0" encoding="UTF-8"?' ];
parser.dtdNodeList = [ '!DOCTYPE MyAppConfig SYSTEM "/dtds/AppConfig.dtd"' ];

console.log( parser.compose() );
```
